test/sqllogictest/scalar_subqueries_select_list.slt

```diff
@@ -801,23 +801,31 @@ Explained Query:
         ArrangeBy keys=[[#0, #1]] // { arity: 3 }
           Union // { arity: 3 }
             Get l3 // { arity: 3 }
-            Map (null) // { arity: 3 }
-              Union // { arity: 2 }
-                Negate // { arity: 2 }
-                  Distinct project=[#0, #1] // { arity: 2 }
-                    Project (#0, #1) // { arity: 2 }
-                      Get l3 // { arity: 3 }
-                Get l1 // { arity: 2 }
+            Project (#0, #1, #4) // { arity: 3 }
+              Map (null) // { arity: 5 }
+                Join on=(#0 = #2 AND #1 = #3) type=differential // { arity: 4 }
```

Comment:
Looks like semijoin elimination didn't happen, potentially because we changed the projections atop `l3` and that transform is not equivalence-aware.